### DISCUSSION TITLE
fail2ban running in separate LXC container from all other VMs and CTs, how do I feed it the logs to scan?

### DISCUSSION THREAD

**Discussion by taydy00** [fail2ban/fail2ban#3667] [General]
Hey all, I am running proxmox for my hypervisor. I set up fail2ban in the same LXC container as Caddy (for reverse proxying). Caddy handles reverse proxying for Home Assistant, Nextcloud, and a couple other small things. I want to set up fail2ban in this container so it stays grouped with my reverse proxy, but it obviously can't see the apache logs from Nextcloud (as it is in a separate container), or the log files from Home Assistant (as it is in a separate VM). Setting up Caddy and fail2ban is all relatively new to me, so I may be missing something quite obvious. How can I get fail2ban to protect these separate containers? Will Caddy log that repeated authentication failures have been happening to Nextcloud/Home Assistant, for example, because it's proxying all the traffic to/from them? Thank you in advance, and I apologize for being so new to this!

**Comment by sebres** [comment]
You would probably need to share the logs with the container where you have fail2ban.
For example via samba, or mount point (e. g. see https://gist.github.com/julianlam/07abef272136ea14a627).

> Will Caddy log that repeated authentication failures have been happening to Nextcloud/Home Assistant, for example, because it's proxying all the traffic to/from them?

It depends. You can try to make few authentication attempts and see what happens in the log.
Additional issue may be the real IP vs. proxying IP in the logged message - for instance if 401 is logged with the IP of the proxying side, but you'd need to ban the IP from `X-Forwarded-For` or `X-Real-IP` headers instead (otherwise you'd ban the proxy instead of the intruder behind the proxy).

**Comment by taydy00** [comment]
Perfect. Alternatively, would it be better to just set up fail2ban on each individual machine? Or does that lead to the same problem, with fail2ban banning the proxy instead of the actual external IP?

**Reply by sebres** [reply]
Again, it depends on the logging behavior of corresponding service (e. g. what exactly it'd log).
Normally every constellation is imaginable, so it is impossible to predict how it is affected by proxying and/or what is possible to do.
Basically single fail2ban instance is enough, but one'd need to configure the jails (logpath, filters and banning actions) properly.

**Reply by taydy00** [reply]
Perfect. I apologize for not replying to your comment, I'm still relatively new to using GitHub for comments/questions. For now I have set it up directly within the container itself, because it's easier to manage if something breaks, but I will look into creating mount points within containers as well. Thank you for your help!
